lib/RTCSession.js

```diff
@@ -1924,7 +1927,10 @@ module.exports = class RTCSession extends EventEmitter
       .then(() =>
       {
         // Resolve right away if 'pc.iceGatheringState' is 'complete'.
-        if (connection.iceGatheringState === 'complete' && (!constraints || !constraints.iceRestart))
+        // Also resolve if the state is still 'gathering' but has already been interrupted via ready().
+        if ((connection.iceGatheringState === 'complete' ||
+          (this._iceReady && connection.iceGatheringState === 'gathering')) &&
+          (!constraints || !constraints.iceRestart))
```

Comment:
```suggestion
        /**
         * Resolve right away if:
         * - 'connection.iceGatheringState' is 'complete' and no 'iceRestart' constraint is set.
         * - 'connection.iceGatheringState' is 'gathering' and 'iceReady' is true.
         */
        const iceRestart = constraints && constraints.iceRestart;
        
        if ((connection.iceGatheringState === 'complete' && !iceRestart) ||
          (connection.iceGatheringState === 'gathering' && this._iceReady))
```

Reply:
`constratins.iceRestart` needs to still be considered. You removed it if `connection.iceGatheringState === 'complete'`